src/common/runtime/src/runtime_throttle_count_mode.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Debug;
+use std::future::Future;
+use std::sync::Arc;
+
+use ratelimit::Ratelimiter;
+use tokio::runtime::Handle;
+pub use tokio::task::JoinHandle;
+
+use crate::future_throttle_count_mode::ThrottleFuture;
+use crate::runtime::{Dropper, RuntimeTrait};
+use crate::throttle_priority::Priority;
+use crate::Builder;
+
+pub struct RuntimeThrottleShareWithFuture {
+    pub ratelimiter: Option<Ratelimiter>,
+}
+
+impl Debug for RuntimeThrottleShareWithFuture {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("RuntimeThrottleShareWithFuture")
+            .field(
+                "ratelimiter_max_tokens",
+                &self.ratelimiter.as_ref().map(|v| v.max_tokens()),
+            )
+            .field(
+                "ratelimiter_refill_amount",
+                &self.ratelimiter.as_ref().map(|v| v.refill_amount()),
+            )
+            .finish()
+    }
+}
+
+/// A runtime to run future tasks
+#[derive(Clone, Debug)]
+pub struct RuntimeThrottleCount {
+    name: String,
+    handle: Handle,
+    share_with_future: Arc<RuntimeThrottleShareWithFuture>,
```

Comment:
This can be more accrue as `shared_with_futures`.

Or it can be as simple as `throttler`